### PR TITLE
Amend rephrase prompt

### PR DESCRIPTION
**Pull request by ggordonhall**
Modify rephrase prompt and remove Rake keyword extraction. Now the rephrase prompt generates a keyword search query directly. This should help address the following issues:
- Rephrase step tries to answer the user question
- Regurgitates answers in the conversational history
- Rephrases are too verbose and complicate search

More work is needed here to improve multi-repo performance.

### CODE REVIEW

File: server/bleep/src/webserver/answer.rs [hunk 1]
```diff
@@ -935,18 +898,17 @@ Assistant:<index>",
         query: &str,
     ) -> api::Messages {
         let system = format!(
-            r#"{}/{}
-=========
-{}
+            r#"{}
 =========
-Above, you have an extract from a code file. This message will be followed by the last few utterances of a conversation with a user. Use the code file to write a concise, precise answer to the question.
+Above, you have an extract from a the {} file in the {} repo. This message will be followed by the last few utterances of a conversation with a user. Use the code file to write a concise, precise answer to the question.
```
[the remark this answers is not in the document]
Comment (nerdypepper) [reply server/bleep/src/webserver/answer.rs:903]: "... a the ..."

File: server/bleep/src/webserver/answer.rs [hunk 2]
```diff
@@ -935,18 +909,17 @@ Assistant:<index>",
         query: &str,
     ) -> api::Messages {
         let system = format!(
-            r#"{}/{}
+            r#"{}
 =========
-{}
-=========
-Above, you have an extract from a code file. This message will be followed by the last few utterances of a conversation with a user. Use the code file to write a concise, precise answer to the question.
+Above, you have an extract from the {} file in the {} repo. This message will be followed by the last few utterances of a conversation with a user. Use the code file to write a concise, precise answer to the question.
```
Comment (nerdypepper) [inline server/bleep/src/webserver/answer.rs:914]: could you double check these string interpolations? we have `snippet.repo_name`, `snippet.relative_path` and `snippet.text` in that order, the prompt looks like it generates this:

```
{repo_name}
=======
Above, you have an extract from the {snippet.relative_path} file in the {snippet.text} repo ...
```

`repo_name` and `text` need to be reversed.

Comment (nerdypepper) [reply server/bleep/src/webserver/answer.rs:914]: it so happens that gpt does not care, but heres a sample prompt:

```
main\n=========\nAbove, you have an extract from the server/bleep/src/repo.rs file in the use dashmap::DashMap;\nuse serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};\nuse std::{\n    fmt::{self, Display},\n    path::{Path, PathBuf},\n    str::FromStr,\n    sync::Arc,\n    time::SystemTime,\n};\nuse tracing::debug;\nuse utoipa::ToSchema;\n\nuse crate::{\n    ctags, indexes,\n    language::{get_language_info, LanguageInfo},\n    state::{get_relative_path, pretty_write_file},\n};\n\npub(crate) type FileCache = Arc<DashMap<PathBuf, FreshValue<String>>>;\n\n#[derive(Serialize, Deserialize)]\npub(crate) struct FreshValue<T> {\n    // default value is `false` on deserialize\n    #[serde(skip)]\n    pub(crate) fresh: bool,\n    pub(crate) value: T,\n}\n\nimpl<T> From<T> for FreshValue<T> {\n    fn from(value: T) -> Self {\n        Self { fresh: true, value }\n    }\n}\n\n// Types of repo\n#[derive(Serialize, Deserialize, ToSchema, Hash, PartialEq, Eq, Clone, Debug)]\n#[serde(rename_all = \"snake_case\")]\npub enum Backend {\n    Local,\n    Github,\n}\n\n// Repository identifier\n#[derive(Hash, Eq, PartialEq, Debug, Clone)]\npub struct RepoRef {\n    pub backend: Backend,\n    pub name: String,\n}\n\nimpl RepoRef {\n    pub fn new(backend: Backend, name: &(impl AsRef<str> + ?Sized)) -> Result<Self, RepoError> {\n        use Backend::*;\n\n        match backend {\n            Github => Ok(RepoRef {\n                backend,\n                name: name.as_ref().to_owned(),\n            }),\n            Local => {\n                let path = Path::new(name.as_ref());\n\n                if !path.is_absolute() {\n                    return Err(RepoError::NonAbsoluteLocal);\n                }\n\n                for component in path.components() {\n                    use std::path::Component::*;\n                    match component {\n                        CurDir | ParentDir => return Err(RepoError::InvalidPath),\n                        _ => continue,\n                    }\n                }\n\n                Ok(RepoRef {\n                    backend,\n                    name: name.as_ref().to_owned(),\n                })\n            }\n        }\n    }\n\n    pub fn from_components(root: &Path, components: Vec<String>) -> Result<Self, RepoError> {\n        let refstr = components.join(\"/\");\n        let pathstr = match refstr.trim_start_matches('/').split_once('/') {\n            Some((\"github.com\", name)) => return RepoRef::new(Backend::Github, name),\n            Some((\"local\", name)) => name,\n            _ => &refstr,\n        };\n\n        let local_path = get_relative_path(Path::new(pathstr), root);\n        Self::new(Backend::Local, &local_path.to_string_lossy())\n    }\n\n    pub fn backend(&self) -> Backend {\n        self.backend.clone()\n    }\n\n    pub fn name(&self) -> &str {\n        &self.name\n    }\n\n    pub fn is_local(&self) -> bool {\n        self.backend == Backend::Local\n    }\n\n    pub fn indexed_name(&self) -> String {\n        // Local repos indexed as: dirname\n        // Github repos indexed as: github.com/org/repo\n        match self.backend {\n            Backend::Local => Path::new(&self.name)\n                .file_name()\n                .expect(\"last component is `..`\")\n                .to_string_lossy()\n                .into(),\n            Backend::Github => format!(\"{}\", self),\n        }\n    }\n\n    pub fn display_name(&self) -> String {\n        match self.backend {\n            // org_name/repo_name\n            Backend::Github => self.name.to_owned(),\n            // repo_name\n            Backend::Local => self.indexed_name(),\n        }\n    }\n\n    pub fn local_path(&self) -> Option<PathBuf> {\n        match self.backend {\n            Backend::Local => Some(PathBuf::from(&self.name)),\n            _ => None,\n        }\n    }\n}\n\nimpl AsRef<RepoRef> for RepoRef {\n    fn as_ref(&self) -> &RepoRef {\n        self\n    }\n}\n\nimpl<P: AsRef<Path>> From<&P> for RepoRef {\n    fn from(path: &P) -> Self {\n        assert!(path.as_ref().is_absolute());\n        RepoRef {\n            backend: Backend::Local,\n            name: path.as_ref().to_string_lossy().to_string(),\n        }\n    }\n}\n\nimpl From<&str> for RepoRef {\n    fn from(refstr: &str) -> Self {\n        Self::from_str(refstr).unwrap()\n    }\n}\n\nimpl FromStr for RepoRef {\n    type Err = RepoError;\n\n    fn from_str(refstr: &str) -> Result<Self, Self::Err> {\n        match refstr.trim_start_matches('/').split_once('/') {\n            // github.com/...\n            Some((\"github.com\", name)) => RepoRef::new(Backend::Github, name),\n            // local/...\n            Some((\"local\", name)) => RepoRef::new(Backend::Local, name), repo. This message will be followed by the last few utterances of a conversation with a user. Use the code file to write a concise, precise answer to the question.\n\n- Format your response in GitHub Markdown. Paths, function names and code extracts should be enclosed in backticks\n- Use markdown bullet points to format lists\n- Keep your response short. It should only be a few sentences long at the most\n- Do NOT copy long chunks of code into the response\n- If the file doesn't contain enough information to answer the question, or you don't know the answer, just say \"Sorry, I'm not sure.\"\n- Do NOT try to make up an answer or answer with regard to information that is not in the file\n- The conversation history can provide context to the user's current question, but sometimes it contains irrelevant information. IGNORE information in the conversation which is irrelevant to the user's current question\n\nLet's think step by step. First carefully refer to the code above, then answer the question with reference to it.
```

Comment (ggordonhall) [reply server/bleep/src/webserver/answer.rs:914]: Fixed!

**Review by nerdypepper — APPROVED** [review]

**Pull request merged**